### PR TITLE
24437 rm line multi plot

### LINKED ISSUE #24437: remove lines in multi plotting
this will allow the user to remove lines from a plot but will not auto remove empty plots.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The multiplotting now includes vertical lines, annotations and the ability to remove lines. 

**To test:**
Open the multi plotting interface (have to enable it first)
You will see some vertical lines and annotations
Click `remove line`
You should be able to remove everything excpet for the line and annotation marked as protected.

Fixes #24437. 

Does not need to be in the release notes. 
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Matthew-Andrew — COMMENTED** [review]
This works for me in python2. When I tried the window on a python3 build however I get an error message saying `dictionary changed size during iteration`. When I try to open the window which then does not open.

File: scripts/test/MultiPlotting/Subplot_test.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        self.subplot._context.subplots["one"] = mock.Mock()
+        self.setup_rm()
+        self.subplot._rm()
+
+        self.assertEquals(self.subplot._raise_rm_window.call_count,1)
+        self.assertEquals(self.subplot._raise_selector_window.call_count,0)
+        self.assertEquals(self.subplot._get_rm_window.call_count,0)
+        self.assertEquals(self.subplot._createSelectWindow.call_count,0)
+
+
```
Comment (Mantid-Matthew) [inline scripts/test/MultiPlotting/Subplot_test.py]: not sure whether this is a problem or not as flake8 is not run on test files but the python standard specifies a one line gap between functions in a class.

**Review by Matthew-Andrew — DISMISSED** [review]
This looks good to me

**Review by gvardany — CHANGES_REQUESTED** [review]
I see some python files follow lower case + underscore naming scheme, some others camel case, or even camel case + underscore. I think it would be more coherent if we have only lowercase and underscores in file names of python scripts.

**Review by Matthew-Andrew — APPROVED** [review]

**Review by gvardany — COMMENTED** [review]
Thanks for renaming. Judging from the diff `scripts/MultiPlotting/subplot/subPlot.py` is not moved to `scripts/MultiPlotting/subplot/sub_plot.py`, but rather is deleted, and a new file is added. I think like this you are losing the history. Can you check what happened? For example, for `subplot_context` the move seems to be done properly.

**Comment by AnthonyLim23** [comment]
I think that I may have done a normal `mv` and not a `git mv`

**Comment by gvardany** [comment]
That's what I thought, but actually in the diff of the last commit, it shows up correctly, but in the diff of the whole PR, it's not.

**Comment by AnthonyLim23** [comment]
huh strange. It should have a rather short history anyway.

**Comment by gvardany** [comment]
Yes I think it's a detail anyhow, just to keep in mind for the future.

**Review by gvardany — APPROVED** [review]

**Pull request merged**